src/main/java/app/nzyme/core/events/actions/EventActionUtilities.java

```diff
@@ -42,6 +44,14 @@ public static EventActionDetailsResponse eventActionEntryToResponse(EventActionE
             ));
         }
 
+        List<DetectionEventTypeDetailsResponse> detectionEventSubscriptions = Lists.newArrayList();
+        for (DetectionType subscription : subscribedDetectionEvents) {
+            detectionEventSubscriptions.add(DetectionEventTypeDetailsResponse.create(
+                    subscription.name(),
+                    subscription.getTitle(),
+                    subscription.getSubsystem().name()
+            ));
+        }
```

Comment:
Have seen you using this pattern a few times. You can do this:

`List<DetectionEventTypeDetailsResponse> detectionEventSubscriptions = subscribedDetectionEvent.stream().
                map(o -> DetectionEventTypeDetailsReponse.create(o.name(), o.getTitle(), o.getSubsystem().name()))
                .collect(Collectors.toList());`

No need to change all these, but might save you some verbosity in the future.

Reply:
for loop perform better and uses less memory for small lists. stream can be parallelized automatically by the vm, so good for large or "endless" datasets.

Reply:
:+1: